subprojects/plugins/src/integTest/groovy/org/gradle/groovy/GroovyCrossCompilationIntegrationTest.groovy

```diff
@@ -19,84 +19,61 @@ package org.gradle.groovy
 import org.gradle.api.JavaVersion
 import org.gradle.integtests.fixtures.AvailableJavaHomes
 import org.gradle.integtests.fixtures.MultiVersionIntegrationSpec
-import org.gradle.integtests.fixtures.TargetVersions
-import org.gradle.internal.jvm.JavaInfo
-import org.gradle.test.fixtures.file.ClassFile
-import org.gradle.util.internal.TextUtil
+import org.gradle.integtests.fixtures.TargetCoverage
+import org.gradle.integtests.fixtures.jvm.JavaToolchainFixture
+import org.gradle.testing.fixture.GroovyCoverage
+import org.gradle.util.internal.VersionNumber
 import org.junit.Assume
 
-@TargetVersions(["1.6", "1.7", "1.8"])
-class GroovyCrossCompilationIntegrationTest extends MultiVersionIntegrationSpec {
-    JavaVersion getJavaVersion() {
-        JavaVersion.toVersion(MultiVersionIntegrationSpec.version)
-    }
+import static org.gradle.util.internal.GroovyDependencyUtil.groovyModuleDependency
 
-    JavaInfo getTarget() {
-        return AvailableJavaHomes.getJdk(javaVersion)
-    }
+@TargetCoverage({ GroovyCoverage.SINCE_3_0 })
+class GroovyCrossCompilationIntegrationTest extends MultiVersionIntegrationSpec implements JavaToolchainFixture {
```

Comment:
🤔 It looks like this test could be merged into `GroovyCompileToolchainIntegrationTest`.

Reply:
Yep, it now has the same scaffolding. Moved it there.